rerender.py

```diff
@@ -0,0 +1,153 @@
+import os
+import json
+import logging
+import pprint
+import tempfile
+import subprocess
+
+import urllib3.util.retry
+
+from github import Github
+from git import Repo, GitCommandError
+
+LOGGER = logging.getLogger(__name__)
+
+
+def create_api_sessions(github_token):
+    """Create API sessions for GitHub.
+
+    Parameters
+    ----------
+    github_token : str
+        The GitHub access token.
+
+    Returns
+    -------
+    gh : github.MainClass.Github
+        A `Github` object from the PyGithub package.
+    """
+    gh = Github(
+        github_token,
+        retry=urllib3.util.retry.Retry(total=10, backoff_factor=0.1))
+
+    return gh
+
+
+def rerender(git_repo):
+    LOGGER.info('rerendering')
+    curr_head = git_repo.active_branch.commit
+    ret = subprocess.call(
+        ["conda", "smithy", "rerender", "-c", "auto"],
```

Comment:
Technically, following should same some time. (But the way it is now is that it just warns after making the solve, but we can fix that in conda-smithy)
```suggestion
        ["conda", "smithy", "rerender", "-c", "auto", "--no-check-uptodate"],
```